helper/file.php

```diff
@@ -153,4 +157,26 @@ protected function execute($dir, $src_ns, $src_name, $dst_ns, $dst_name, $extreg
         }
         return true;
     }
+
+    /**
+     * Edit `.changes` meta file in order to make history of old entries accessible.
+     * 
+     * if pagename entry is not edited, users can't read or diff old entries that edited before moving.
+     *
+     * @param string $old_path Old `.changes` file path.
+     * @param string $new_path New `.changes` file path.
+     */
+    protected function edit_changes_file($old_path, $new_path) {
+        global $conf;
+
+        $history_content = file_get_contents($new_path);
+
+        $old_pagename = str_replace("/", ":", substr($old_path, strlen($conf['metadir']) + 1, -8));
+        $new_pagename = str_replace("/", ":", substr($new_path, strlen($conf['metadir']) + 1, -8));
```

Comment:
I think this is missing an `utf8_decodeFN` to remove the (potential) encoding of the filename. Also, what speaks against just using `$src_ns`, `$dst_ns`, `$src_name` and `$dst_name` in the calling method to construct both IDs without the complicated step of first constructing filenames?